AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/StateMachine/Resolvers/SignOut/SignOutState+Resolver.swift

```diff
@@ -87,14 +92,19 @@ extension SignOutState {
             }
         }
 
-        private func resolveClearingCredentialStore(byApplying signOutEvent: SignOutEvent,
+        private func resolveClearingCredentialStore(byApplying event: StateMachineEvent,
                                                     from oldState: SignOutState,
-                                                    signedInData: SignedInData) -> StateResolution<SignOutState> {
-            switch signOutEvent.eventType {
-            case .signedOutSuccess:
+                                                    signedInData: SignedInData)
+        -> StateResolution<SignOutState> {
+            guard let authEvent = event.isAuthEvent else {
+                return .from(oldState)
+            }
```

Comment:
Should we handle AuthEvents in only in the Auth Resolver?

I am asking this because how would we know when to pass the events to a child resolver?

Reply:
We should pass in all events to sub states no matter what type it is. 
We could do that above logic with specific event type, but that will unnecessary create new events and new actions.